doc/developer-guide/api/functions/TSUrlStringGet.en.rst

```diff
@@ -61,6 +62,31 @@ terminator - :arg:`length` must be used to correctly display the string.
 This function is useful to guarantee a URL that is as complete as possible given the specific
 request.
 
+:func:`TSHttpHdrEffectiveUrlBuffGet` returns the effective URL for any HTTP request (not just the client request).
+If the request has a Host header field (and the URL does not contain a host specifier), the host specifier the header
+provides is inserted into the URL.  The host and
+scheme in the returned URL will be normalized to lower case letters (to make URL comparisons simple and fast).
+This prints the effective URL for the header specified by :arg:`hdr_buf` and
+:arg:`hdr_loc` to the buffer starting at :arg:`buff`. If the effective URL is longer than :arg:`size`, nothing is
+written to :arg:`buff`.  Note that this is not considered an error case, the function will still reeturn `TS_SUCCESS`.
+It is the responsibility of the caller to check this result to determine if output was generated.
+The full length of the URL is always returned in :arg:`*length` when the function returns `TS_SUCCESS`.
+
+The typical usage would be
+::
+
+  TSMBuffer hdr_buff;
+  TSMLoc hdr_loc;
+  TSHttpTxnServerReqGet(txn, &hdr_buff, &hdr_loc);
+  int64_t length;
+  char store[2048];
+  char *buff = store;
+  TSHttpHdrEffectiveUrlBuffGet(hdr_buff, hdr_loc, buff, sizeof(store), &length);
+  if (length > sizeof(store)) {
+    buff = malloc(length);
```

Comment:
buff is 2 different types in this example.

Reply:
Does it matter since it will convert implicitly:  https://godbolt.org/z/EMfhaj

Reply:
That example doesn't address the conversion of a function returning `void *` to a variable of type `char *`. I'm surprised this works, there shouldn't be an implicit conversion of that.